application/src/main/java/com/sanction/lightning/resources/TwitterResource.java

```diff
@@ -217,14 +220,45 @@ public Response getOAuthUrl(@Auth Key key,
 
     TwitterService service = twitterServiceFactory.newTwitterService();
 
-    String url = service.getAuthorizationUrl(redirectUrl);
-    if (url == null) {
+    TwitterOAuthRequest oauthRequest = service.getAuthorizationUrl(redirectUrl);
+
+    if (oauthRequest == null) {
       LOG.error("Unable to build OAuth URL for Twitter.");
       return Response.status(Response.Status.SERVICE_UNAVAILABLE)
           .entity("Unable to retrieve OAuth URL from Twitter.").build();
     }
 
     LOG.info("Successfully built OAuth URL for Twitter.");
-    return Response.ok(url).build();
+    LOG.info("Token: {}", oauthRequest);
+    return Response.ok(oauthRequest).build();
```

Comment:
This endpoint would now instead of returning just a URL string, return a JSON with the URL, request token, and request secret.